### PR TITLE
refactor: make `pubkey` optional for `decoded_json` in `keystore` in `decode_str!` function

### LINKED ISSUE #1084: Make `pubkey` field optional
Once #1067 is merged, we can make the `pubkey` field optional on keystores. For that, we'd need to change this `fetch!` here:

https://github.com/lambdaclass/lambda_ethereum_consensus/blob/5160683ae3834fb3eb45daaa556b46e73d2dd544/lib/keystore.ex#L27

### PR DESCRIPTION
**Pull request by artemd24**
**Motivation**

There is not need to raise exception because of cheking pub keys [further](https://github.com/lambdaclass/lambda_ethereum_consensus/blob/ce09e953fe6fd12407fc0ec7b733e90e13d16b7c/lib/keystore.ex#L31)

**Description**

Change `Map.fetch!(decoded_json, "pubkey") |> parse_binary!()` to 

```Elixir
pubkey =
      case Map.has_key?(decoded_json, "pubkey") do
        true -> Map.get(decoded_json, "pubkey") |> parse_binary!()
        false -> derived_pubkey
      end
```

+some tests added

Resolves #1084

Closes #1084

### CODE REVIEW

**Review by MegaRedHand — COMMENTED** [review]
Hey there @artemd24! Thanks for your contribution!

As part of this issue, we need to handle when the "pubkey" field isn't specified. In that case, we need to ignore the following comparison against `derived_pubkey`, and set `pubkey = derived_pubkey`. Also, please add unit tests to this change (you can find examples in [`test/unit/keystore_test.exs`](https://github.com/lambdaclass/lambda_ethereum_consensus/blob/ce09e953fe6fd12407fc0ec7b733e90e13d16b7c/test/unit/keystore_test.exs#L74)).

**Comment by artemd24** [comment]
Hello @MegaRedHand!

I have actualized changes according to your comment.
If you have more improvement ideas, please write it, I'll make it up with pleasure!

File: test/unit/keystore_test.exs [hunk 1]
```diff
@@ -92,4 +92,34 @@ defmodule Unit.KeystoreTest do
     {:ok, signature} = Bls.sign(privkey, digest)
     assert Bls.valid?(pubkey, digest, signature)
   end
+
+  test "eip scrypt without pubkey test vector" do
+    decoded_json = Jason.decode!(@scrypt_json)
+    Map.delete(decoded_json, "pubkey")
```
Comment (MegaRedHand) [inline test/unit/keystore_test.exs:98]: `Map.delete` returns the modified map. The json with the deleted pubkey is being discarded here.

File: test/unit/keystore_test.exs [hunk 2]
```diff
@@ -92,4 +92,34 @@ defmodule Unit.KeystoreTest do
...
+
+    digest = :crypto.hash(:sha256, "test message")
+    {:ok, signature} = Bls.sign(privkey, digest)
+    assert Bls.valid?(pubkey, digest, signature)
+  end
+
+  test "eip pbkdf2 without pubkey test vector" do
+    decoded_json = Jason.decode!(@pbkdf2_json)
+    Map.delete(decoded_json, "pubkey")
+    pbkdf2_json = Jason.encode!(decoded_json)
```
Comment (MegaRedHand) [inline test/unit/keystore_test.exs:114]: Same here. Also, you can call multiple functions one after the other with pipes:
```suggestion
    pbkdf2_json =
      Jason.decode!(@pbkdf2_json)
      |> Map.delete("pubkey")
      |> Jason.encode!()
```

Comment (artemd24) [reply test/unit/keystore_test.exs]: Wow, looks great! 
I have done this following your example

File: test/unit/keystore_test.exs (hunk 1 above)
Comment (artemd24) [reply test/unit/keystore_test.exs]: Done

**Comment by MegaRedHand** [comment]
Great work @artemd24 ! There seem to be some formatting errors. Please run `mix format`.

**Comment by artemd24** [comment]
@MegaRedHand thank you for helping me! 
If you have more good first issues I will be glad to contribute

**Review by MegaRedHand — APPROVED** [review]
Thanks again for your contribution! 😄

**Comment by MegaRedHand** [comment]
@all-contributors please add @artemd24 for code

**Pull request merged**